libp2p/Host.cpp

```diff
@@ -328,21 +289,17 @@ void Host::runAcceptor()
 	{
 		clog(NetConnect) << "Listening on local port " << m_listenPort << " (public: " << m_tcpPublic << ")";
 		m_accepting = true;
-		m_socket.reset(new bi::tcp::socket(m_ioService));
-		m_tcp4Acceptor.async_accept(*m_socket, [=](boost::system::error_code ec)
+		
+		bi::tcp::socket* s = new bi::tcp::socket(m_ioService);
+		m_tcp4Acceptor.async_accept(*s, [=](boost::system::error_code ec)
```

Comment:
Typo. The last word of prev comment should read 'exiting' and not accepting.
The acceptor is added to asio and when the acceptor is closed, asio will call the lamda with an error indicating acceptor was closed.

I'll see what I can do but for now it may remain allocated and deleted instead of a managed ptr. The reason for this is scope; it's been working for awhile as-is and the proper solution is abstracting boost's acceptor/socket/connect lifecycle into a "Connection". This has been implemented in another branch but present objective is node discovery. The reason I DID change this is because m_socket should not be a property of Host; it needs to be deallocated with respect to the acceptor (this lambda) and not Host.